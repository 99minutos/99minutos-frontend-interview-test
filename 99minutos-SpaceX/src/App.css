* {
  margin: 0;
  font-family: 'Comfortaa', cursive;
}

body { 
  background-color: rgb(4, 7, 49);
}

.container-app {
  background-color: rgb(194, 194, 194);
  margin: 6rem 3rem 5rem 3rem;
}

h1{
  text-align: center;
}

.containerMissions{
  background-color: rgb(194, 194, 194);
}

