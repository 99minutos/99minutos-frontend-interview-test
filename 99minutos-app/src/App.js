import React from "react";
import {BrowserRouter} from 'react-router-dom';
import LastLaunches from "./components/LastLaunches/LastLaunches";

function App() {
  return (
   <BrowserRouter>
   <LastLaunches/> 
  </BrowserRouter>
  );
}

export default App;