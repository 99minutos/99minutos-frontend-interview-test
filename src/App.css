main{
  width: 80%;
  height: 27.5rem;
  display: flex;
  margin: auto;
  overflow: hidden;
  padding: 1rem;
}

section{
  width: 35%;
  height: auto;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 0.3rem;
  overflow: scroll;
  box-sizing: border-box;
}

.mission{
  width: 65%;
  padding: 0.3rem;
  text-align: center;
}

.mission div{
  width: 100%;
  height: 15.6rem;
  box-sizing: content-box;
  overflow: hidden;
  display: flex;
  justify-content: center;
  align-items: center;
}

.mission img {
  width: 100%;
}

.mission h2{
  font-size: 1.3rem;
  margin: 0.2rem;
}

.mission p{
  font-size: 11px;
}

/* ===== Buttons  ===== */

button{
  font-size: 0.7rem;
  border: none;
  border-radius: 0.6rem;
  padding: 0.5rem;
  margin: 0.2rem;
  color: #fff;
  background: #131313;
}

button:hover{
  color: #f87966;
  transition: 0.2s;
  cursor: pointer;
}

.btnMission{
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
}

.btnMission:focus{
  background: #8f54a0;
}

.btnLine{
  border: 1px solid #f87966;
}

h1,h2{
  text-align: center;
  font-weight: bold;
}

h1{
  margin: 0;
}

h2{
  margin: 0.5rem;
}

/* ===== Card Item  ===== */

.cardItem{
  width: 90%;
  height: 20%;
  display: flex;
  align-items: center;
  margin: 0.4rem;
}

.cardItem img{
  margin: 1rem;
}

.cardItem h3, .cardItem p {
  margin: 0.2rem;
  text-align: left;
}

.cardItem h3{
  font-size: 0.7rem;
}

.cardItem p{
  font-size: 0.6rem;
}

.cardItem #date{
  font-size: 0.6rem;
  text-align: right;
  margin-top: 0.3rem;
  color: #cacaca;
}

/* ===== Scrollbar CSS ===== */
  /* Firefox */
  * {
    scrollbar-width: auto;
    scrollbar-color: #8f54a0 #000;
  }

  /* Chrome, Edge, and Safari */
  *::-webkit-scrollbar {
    width: 16px;
  }

  *::-webkit-scrollbar-track {
    background: #000;
  }

  *::-webkit-scrollbar-thumb {
    background-color: #8f54a0;
    border-radius: 10px;
    border: 3px solid #000;
  }

  *::-webkit-scrollbar-corner {
    background: transparent;
  }