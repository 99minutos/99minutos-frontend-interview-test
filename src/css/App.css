/* Reset CSS */
html, body, div, span, applet, object, iframe,
h1, h2, h3, h4, h5, h6, p, blockquote, pre,
a, abbr, acronym, address, big, cite, code,
del, dfn, em, img, ins, kbd, q, s, samp,
small, strike, strong, sub, sup, tt, var,
b, u, i, center,
dl, dt, dd, ol, ul, li,
fieldset, form, label, legend,
table, caption, tbody, tfoot, thead, tr, th, td,
article, aside, canvas, details, embed, 
figure, figcaption, footer, header, hgroup, 
menu, nav, output, ruby, section, summary,
time, mark, audio, video {
	margin: 0;
	padding: 0;
	border: 0;
	font-size: 100%;
	font: inherit;
	vertical-align: baseline;
}
/* HTML5 display-role reset for older browsers */
article, aside, details, figcaption, figure, 
footer, header, hgroup, menu, nav, section {
	display: block;
}
body {
	line-height: 1;
}
ol, ul {
	list-style: none;
}
blockquote, q {
	quotes: none;
}
blockquote:before, blockquote:after,
q:before, q:after {
	content: '';
	content: none;
}
table {
	border-collapse: collapse;
	border-spacing: 0;
}

/* CSS CUSTOM PROPERTIES BEGIN */

:root {
  --color-white-primary: #F3F3F3;
  --color-black-primary: #060503;

  --body-font: 'Roboto', sans-serif;
  --font-size-large: 28px;
  --font-size-medium: 22px;
}

/* CSS CUSTOM PROPERTIES END */

body {
  padding: 20px;
  background-color: var(--color-black-primary);
  font-family: var(--body-font);
  color: var(--color-white-primary);
  background-image: url('../images/space.jpeg');
}

h1 {
  font-size: var(--font-size-large);
  font-weight: bold;
  margin-bottom: 20px;
}

h2 {
  font-size: var(--font-size-medium);
  font-weight: bold;
}

.Error {
  text-align: center;
}