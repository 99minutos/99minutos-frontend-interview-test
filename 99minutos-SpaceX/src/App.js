import React, { useState } from 'react'
import './App.css';
import { gql, useQuery } from '@apollo/client'
import ListMissions from './components/ListMissions';
import Mission from './components/Mission';

const launches_past = gql`
query {
    launchesPast(limit: 10) {
        mission_name
        launch_date_local
        details
        launch_site {
          site_name_long
        }
        links {
            article_link
            flickr_images
        }
    }
}
`

function App() {

  const {data, error, loading} = useQuery(launches_past)

  const [ selectedMission, setSelectedMission ] = useState(null)

  if(error) return <span style='color: red'>{error}</span>

  return (
      <div className='container-app row '> 
        <div  className="containerMissions card col-12 col-sm-6 col-xl-5 p-0 m-0">
          <h1 className='m-3'>Last Launches</h1>
          { loading 
            ? <div class="d-flex justify-content-center">
                <div class="spinner-border" role="status">
                  <span class="visually-hidden">Loading...</span>
                </div>
              </div>
            : 
                <div className='listMissions'><ListMissions setSelectedMission={setSelectedMission}  mission={data?.launchesPast} /></div>            
          }
        </div>
        <div className='MissionDetails col-12 col-sm-6 col-xl-7 p-0 m-0'><Mission selectedMission={selectedMission}/></div>
      </div>
  );
}

export default App;
