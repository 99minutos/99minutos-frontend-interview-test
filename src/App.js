import ButtonContainer from "./components/ButtonContainer";
import "./css/styles.css";

function App() {
  return (
    <>
      <div className="app">
        <ButtonContainer />
      </div>
    </>
  );
}

export default App;
