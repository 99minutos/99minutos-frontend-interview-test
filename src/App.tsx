import React from 'react';
import './App.css';
import { HomeScreen } from './pages/home/HomeScreen';

const App = () => {
  	return (
    
    	<HomeScreen/>

  	);
}

export default App;
