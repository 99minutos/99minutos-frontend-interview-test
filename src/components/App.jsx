import React, { Fragment, useState, useEffect } from 'react';
import '../css/App.css';
import Card from "./Card";
import Modal from "./Modal";

const App = () => {
  const [isOpen, setIsOpen] = useState(false)

  const [missions, setMissions] = useState([]);
  const [missionSelected, setMissionSelected] = useState({});
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState('');

  useEffect(() => {
    async function getData() {
      try{
        const response = await fetch('https://api.spacex.land/graphql/', {
          method: 'POST',
          headers: {
            'Content-Type': 'application/json',
            'Accept': 'application/json',
          },
          body: JSON.stringify({
            query: `{
              launches(limit: 10) {
                id
                mission_id
                mission_name
                launch_date_local
                details
                links {
                  flickr_images
                  video_link
                }
              }
            }`
          })
        });
        setLoading(false);
        const body = await response.json();
        setMissions(body.data.launches);
      } catch(error) {
        setError('Something went wrong')
      }
    };
    getData()
  }, [])

  const handleClick = (event, key) => {
    setMissionSelected(missions[key])
    setIsOpen(!isOpen)
  }
  return (
    <Fragment>
      <h1>Last Launches ðŸš€</h1>
      {!loading && !error && missions.map((mission, index) => {
        return <Card
          handleClick={(event) => handleClick(event, index)}
          missionName={mission.mission_name}
          description={mission.details}
          launchDateLocal={mission.launch_date_local}
          key={mission.id}
        />
      })}
      {error && <h1 className="Error">{error}</h1>}
      <Modal onClose={handleClick} isOpen={isOpen} missionSelected={missionSelected} />
    </Fragment>
  );
};

export default App;