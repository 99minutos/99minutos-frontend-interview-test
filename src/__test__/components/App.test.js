import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import App from '../../components/App';

describe('<App />', () => {
  const app = shallow(<App />);
  test('Component Render App', () => {
    expect(app.length).toEqual(1);
  });
});

describe('App Snapshot', () => {
  test('Verify the component App UI', () => {
    const app = create(<App />);
    expect(app.toJSON()).toMatchSnapshot();
  });
});