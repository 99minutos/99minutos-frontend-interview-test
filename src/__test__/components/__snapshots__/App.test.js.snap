// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot Verify the component App UI 1`] = `
Array [
  <h1>
    Last Launches ðŸš€
  </h1>,
  "",
]
`;
